hack/goreleaser.sh

```diff
@@ -22,6 +22,9 @@
 set -o errexit
 set -o pipefail
 
+IMAGE_REGISTRY="${IMAGE_REGISTRY:-ghcr.io/infracloudio}"
```

Comment:
I am not very comfortable with pushing images to `ghcr.io/infracloudio` on every PR.
How are we going to safeguard this against the abuse? People can raise fake PRs with the intent to steal sensitive info like GitHub creds in this case.
Ideally, I would have liked it if the image gets pushed to the PR authors repo rather than `ghcr.io/infracloudio/botkube`

Reply:
there is such threat but it was mitigated with the usage of:

- permissions
  ```yaml
      permissions:
        contents: read
  ```
- and stopping creds injection
  ```yaml
      steps:
        - name: Checkout code
          uses: actions/checkout@v2
          with:
            ref: ${{ github.event.pull_request.head.sha }}
            persist-credentials: false
  ```

In this way, the `save-image`  job which checks out untrusted source code doesn't have any creds.

The `push-image` has necessary perms, but it checks out code from the `develop` branch, which is the trusted one.

Here is also the alternative approach: https://github.com/infracloudio/botkube/pull/604

There is also a discussion which one is better.

Reply:
> In this way, the save-image job which checks out untrusted source code doesn't have any creds.

@mszostok Does that mean the image push won't happen for PRs from the forked repo? What does it mean by untrusted source code?

Reply:
Here is a short introduction to the problem that I recorded previously: https://www.loom.com/share/2f0fdc621a144d32861d83946bfff1d7

In general, we have two options:
- push the image only when there is a given label on a PR, see: https://github.com/infracloudio/botkube/pull/604
- push the image automatically for each new PR, see this PR

Both of those approaches:
1. Build the container image with the changes added on a PR.
2. Push the container image to `ghcr.io/infracloudio/pr`.  


> What does it mean by untrusted source code?

It's a code that was changed/added by contributor on a PR.

Reply:
the label-based approach sounds good to me - As it would only build images only when someone needs them. And repo collaborators also have some kind of control over it

Reply:
@PrasadG193 It would be good to run true E2E tests for each PR, which could make the automatic approach more convenient. I'm working on E2E tests which installs BotKube Helm chart on a local K3d cluster, and then we can run tests against a given test Slack workspace.

Reply:
@PrasadG193 It would be good to run true E2E tests for each PR (that use the built Docker images), which could make the automatic approach more convenient. I'm working on E2E tests which installs BotKube Helm chart on a local K3d cluster, and then we can run tests against a given test Slack workspace. I think we should consider also that point as well. It would be great if you could read the discussion in https://github.com/infracloudio/botkube/pull/604 and see my some concerns. Generally the automatic approach is safer and less error-prone, and it could be run before reviewing PR (there is no way to override PR images from an open PR). In the label approac,  a person is responsible for labeling the PR after code review - as there might be some changes which overrides current images even before merging the PR.